### ISSUE TITLE
--ignore sbref doesn't seem to work as expected

### ISSUE THREAD

**Issue by julfou81** [nipreps/fmriprep#2369]
### What version of fMRIPrep are you using?

fMRIPrep version: 20.2.1 

### What kind of installation are you using? Containers (Singularity, Docker), or "bare-metal"?

singularity on HPC

### What is the exact command-line you used?

 ``` /usr/local/miniconda/bin/fmriprep --fs-license-file /work/freesurfer/license.txt /work/PhantomPain /work/PhantomPain/derivatives participant --participant-label testpilote3 -w /work/temp_data_PhantomPain --mem-mb 50000 --omp-nthreads 10 --nthreads 12 --ignore slicetiming sbref --fd-spike-threshold 0.5 --output-spaces MNI152NLin2009cAsym MNI152NLin6Asym --return-all-components```

### Have you checked that your inputs are BIDS valid?

yes, it does pass the bids-validator

### Did fMRIPrep generate the visual report for this particular subject? If yes, could you share it?

visual report here: https://www.dropbox.com/s/f8awapf73qxflue/export_fmriprep_report_ignore_sbref.zip?dl=0

### Can you find some traces of the error reported in the visual report (at the bottom) or in *crashfiles*?

nothing to declare

### Are you reusing previously computed results (e.g., FreeSurfer, Anatomical derivatives, work directory of previous run)?

I have been re-using FreeSurfer results but took care of removing the temporary files before launching a new execution.

Hi FMRIPREP experts,

I would like to report an observation on the behavior of fmriprep when passing ```--ignore sbref ``` to the command line:
- I was expecting this argument to actually ignore the sbref image when calculating the reference image which is used as a target for HMC. I don't think it is actually the case. Indeed, I am tracking down a misbehavior of mcflirt (see #2360 ) when using SBref as a target for realignment, which I think is maybe related to the difference of contrast between SBref images and the functional run. 
To check that, I relaunched fmriprep while having removed the sbref images from my dataset. Indeed the realignment results were quite different.
Later, I noticed that it is possible to use the ```--ignore sbref``` argument, which looked more convenient that having to remove the SBref images. However, launching fmriprep this way, I obtained the same realignment parameters than with the sbref case, even if the reference image was indeed calculated from the non-steady volumes in the end.

**Comment by effigies** [comment]
Thanks for the report! I think I found the issue, and proposed #2370. If you're able to test in a [development environment](https://fmriprep.org/en/stable/contributors.html#patching-working-repositories) it would be great if you could verify whether it resolves your issue. Might be tricky on Singularity, though...

**Comment by julfou81** [comment]
Thank you for your quick action!  I am not familiar with the development environment. I will check what I can do and will give you a feedback if I get something done to test that.

**Comment by effigies** [comment]
The simplest thing would be to use `fmriprep-docker` on a local machine:

```
pip install fmriprep-docker==20.2.1
fmriprep-docker --help
# Y to download default image
git clone https://github.com/nipreps/fmriprep.git
cd fmriprep
git checkout fix/ignore_sbref
fmriprep-docker --patch fmriprep=fmriprep/ [USUAL ARGS]
```

**Comment by julfou81** [comment]
ok, thank you for help! I will do that and let you know how it goes.

**Comment by julfou81** [comment]
Hi, sorry for the delay, it took me some time to access a computer where I could run docker.
I tried to run the patch you provided and it works now as expected, sbref images are well ignored when the argument ``--ignore sbref`` is passed. This helps when mcflirt with sbref gives a lot of zeros in the motion parameters .par file.

**Comment by effigies** [comment]
Thanks for following up! I've merged it, and the fix will be included in 20.2.2 and 21.0.0.
